### ISSUE TITLE
Impossible to rename a translated category/tag

### ISSUE THREAD

**Issue by herrvigg** [herrvigg/qtranslate-x#383]
<a href="https://github.com/milhouse1337"><img src="https://avatars1.githubusercontent.com/u/326740?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [milhouse1337](https://github.com/milhouse1337)**
_Thursday May 05, 2016 at 21:57 GMT_
_Originally opened as https://github.com/qTranslate-Team/qtranslate-x/issues/383_

----

Hi there,

First thank you for this excellent plugin.

I found a bug in the way categories/tags are translated. In fact it seems that the table `wp_terms` is just a reference to be used with the `qtranslate_term_name` option. Here is a screenshot of the decoded `qtranslate_term_name` it's basically just an array with the translations in it.

![screen shot 2016-05-05 at 5 47 54 pm](https://cloud.githubusercontent.com/assets/326740/15058492/83ba071c-12ea-11e6-9dde-2312416e21d6.png)

The problem is that the key used to match the translation is the category name!! Why ? If the category name changes overtime the translation will never be updated because that key doesn't exist anymore.

To solve this problem, I think the key used should be the term ID. What do you think ?

I can confirm it the bug is present with the latest version of WP and qTranslate-X installed.

Thank you!

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Friday May 06, 2016 at 15:00 GMT_

----

Category name instead of id is how WP uses it mostly. In many cases ID is not even available. I am not sure what problem you refer to. Could you list the steps to reproduce the original problem you are trying to solve?

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Friday May 06, 2016 at 15:31 GMT_

----

Could you mention with a few words what the status is? Does it work for you now?

**Comment by herrvigg** [comment]
<a href="https://github.com/milhouse1337"><img src="https://avatars1.githubusercontent.com/u/326740?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [milhouse1337](https://github.com/milhouse1337)**
_Friday May 06, 2016 at 16:05 GMT_

----

Here is more details. The steps:
- You install qTranslate-X and configure 2 (or more) language, like normal.
- You create a category for the posts named: Test-en (in english)
- You translate it in the 2nd language: Test-fr (in french) -- using the switcher
- You add the category like normal.
- At this point everything is ok.

> You can see in the table `wp_terms` that your category name is in the default language, here: Test-en. There is no [:en]...[:] tags around, because the translation is on the `qtranslate_term_name` as this time.
- You click on edit and change the name to: [:fr]Test-fr-renamed[:en]Test-en-renamed[:]
- You will now have this exact string on the `wp_terms` table `[:fr]Test-fr-renamed[:en]Test-en-renamed[:]` instead of having the category name in the default language.

Basically the category name is not consistent, the first time it's just named using the default language and if you edit it, it's wrapped with the qTranslate-X tags.

Thank you for you help!

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Saturday May 07, 2016 at 04:16 GMT_

----

Are you using Raw Editor Mode? Do you see LSB on category editing page?

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Saturday May 07, 2016 at 23:20 GMT_

----

I did not try exactly your steps yet, but there was other problem #366 resently solved in https://github.com/qTranslate-Team/qtranslate-x/archive/stable.zip and maybe it took care of your problem too. Could you please try?

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Saturday May 07, 2016 at 23:25 GMT_

----

BTW, if you do not see LSB on page `/wp-admin/term.php?taxonomy=xxx&tag_ID=xxx`, then you did not update properly (read bold in https://qtranslatexteam.wordpress.com/2016/04/23/release-notes-3-4-6-5/).

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Sunday May 08, 2016 at 16:01 GMT_

----

Hi Pascal, @milhouse1337, Is this about your case: http://qtranslate-x.com/support/index.php?topic=396.msg2514#msg2514 too?

**Comment by herrvigg** [comment]
<a href="https://github.com/milhouse1337"><img src="https://avatars1.githubusercontent.com/u/326740?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [milhouse1337](https://github.com/milhouse1337)**
_Tuesday May 10, 2016 at 16:46 GMT_

----

@johnclause Thank you for this! I will test and let you know.

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Wednesday Jun 01, 2016 at 15:15 GMT_

----

There is a [pre-release](https://github.com/qTranslate-Team/qtranslate-x/tree/stable) which addresses more isses with terms editing. It would be great if you could test it too.

I am closing the thread, assuming that it is resolved and to save future clicks, we can still write into closed thread or may re-open it if needed.
